support-frontend/assets/components/subscriptionCheckouts/stripeForm/stripeForm.jsx

```diff
@@ -0,0 +1,207 @@
+/* eslint-disable react/no-unused-state */
+// @flow
+
+import React, { Component, type Node } from 'react';
+import { injectStripe } from 'react-stripe-elements';
+import { CardElementWrapper } from './cardElements';
+import Button from 'components/button/button';
+import { ErrorSummary } from '../submitFormErrorSummary';
+import { type FormError } from 'helpers/subscriptionsForms/validation';
+import { type FormField } from 'helpers/subscriptionsForms/formFields';
+import { CardNumberElement, CardExpiryElement, CardCvcElement } from 'react-stripe-elements';
+
+import './stripeForm.scss';
+
+// Types
+
+type PropTypes = {
+  component: Node,
+  stripe: Object,
+  allErrors: FormError<FormField>[],
+  setStripeToken: Function,
+  submitForm: Function,
+  name: string,
+  validateForm: Function,
+}
+
+type StateTypes = {
+  cardNumber: Object,
+  cardExpiry: Object,
+  cardCvc: Object,
+  cardErrors: Array<Object>,
+}
+
+// Styles for stripe elements
+
+const baseStyles = {
+  fontSize: '16px',
+  color: '#000',
+  '::placeholder': {
+    color: 'white',
+  },
+};
+
+const invalidStyles = {
+  color: '#c70000',
+};
+
+// Main component
+
+class StripeForm extends Component<PropTypes, StateTypes> {
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
We should handle all of this state in Redux. That that might also help with persisting the values when the address changes.